osu.Game/Overlays/Mods/ModsEffectDisplay.cs

```diff
@@ -0,0 +1,144 @@
+                    {
+                        RelativeSizeAxes = Axes.Y,
+                        AutoSizeAxes = Axes.X,
+                        ColumnDimensions = new[]
+                        {
+                            new Dimension(GridSizeMode.AutoSize),
+                            new Dimension(GridSizeMode.Absolute, value_area_width)
+                        },
+                        Content = new[]
+                        {
+                            new Drawable[]
+                            {
+                                new Container
+                                {
+                                    RelativeSizeAxes = Axes.Y,
+                                    AutoSizeAxes = Axes.X,
+                                    Masking = true,
+                                    CornerRadius = ModSelectPanel.CORNER_RADIUS,
+                                    Children = new Drawable[]
+                                    {
+                                        labelBackground = new Box
+                                        {
+                                            RelativeSizeAxes = Axes.Both
+                                        },
+                                        new OsuSpriteText
+                                        {
+                                            Anchor = Anchor.Centre,
+                                            Origin = Anchor.Centre,
+                                            Margin = new MarginPadding { Horizontal = 18 },
+                                            Shear = new Vector2(-ShearedOverlayContainer.SHEAR, 0),
+                                            Text = Label,
+                                            Font = OsuFont.Default.With(size: 17, weight: FontWeight.SemiBold)
+                                        }
+                                    }
+                                },
+                                content = new Container
+                                {
+                                    AutoSizeAxes = Axes.Both,
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                    Shear = new Vector2(-ShearedOverlayContainer.SHEAR, 0)
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            labelBackground.Colour = colourProvider.Background4;
+        }
+
+        /// <summary>
+        /// Fades colours of text and its background according to displayed value.
+        /// </summary>
+        /// <param name="difference">Difference between actual value and default value.
+        /// Negative leads to green color, positive to red. Can be obtained via CompareTo(defaultValue).</param>
+        protected void SetColours(int difference)
```

Comment:
Not a huge fan of this API design. I see what you're going for - implementations of this component may have different ideas whether a larger value of their respective metric is, depending on whether it makes gameplay easier or harder - but as is, you have to remember to call `SetColours(0)` manually to get the component into a sane initial state.

A better route may look something like (loose sketch):

```csharp
public class ModsEffectDisplay<TValue> : IHasCurrentValue<TValue>
{
    public Bindable<TValue> Value { get; }

    public enum ModEffect
    {
         NoChange,
         DifficultyReduction,
         DifficultyIncrease
    }

    public abstract ModEffect CalculateEffectFor(TValue value);

    protected override LoadComplete()
    {
        Current.BindValueChanged(current =>
        {
            var effect = CalculateModEffect(current.NewValue);
            setColours(effect);
        }
    }
}
```

Gains from this design:

- Implementations get the `Current` bindable for free
- Implementations don't have to manually manage the colours - all they need to do is implement the pure abstract method that determines what the new value represents difficulty-wise